src/V3Width.cpp

```diff
@@ -121,12 +121,11 @@ std::ostream& operator<<(std::ostream& str, const Castable& rhs) {
 }
 
 #define v3widthWarn(lhs, rhs, msg) \
-    v3errorEnd((V3Error::s().m_mutex.lock(), \
-                V3Error::s().v3errorPrep((lhs) < (rhs)   ? V3ErrorCode::WIDTHTRUNC \
-                                         : (lhs) > (rhs) ? V3ErrorCode::WIDTHEXPAND \
-                                                         : V3ErrorCode::WIDTH), \
-                (V3Error::s().v3errorStr() << msg), V3Error::s().v3errorStr())), \
-        V3Error::s().m_mutex.unlock()
+    v3errorEnd((V3Error::v3errorPrep((lhs) < (rhs)   ? V3ErrorCode::WIDTHTRUNC \
+                                     : (lhs) > (rhs) ? V3ErrorCode::WIDTHEXPAND \
+                                                     : V3ErrorCode::WIDTH) \
+                    << msg, \
+                V3Error::v3errorStr()))
```

Comment:
Isn't this non-equivalent?  In the old version the lock would cover creation of the arguments, e.g. V3Error::v3errorStr, while in the new version the call to v3errorStr will occur outside the lock.

Not quite as good, but perhaps make a static function that wraps V3Error::s().m_mutex.lock() and unlock()?

Reply:
It still covers, as we call the unlock after the message is printed, in V3Error::v3errorEnd, which would be called by :: v3errorEnd/AstNode::v3errorEnd/AstFileLine::v3errorEnd/... It's the epilogue of the message printing.

Reply:
This demonstrates what I mean:

```
#include <iostream>

void lock() { std::cout << "--Lock\n"; }
void unlock() { std::cout << "--Unlock\n"; }

std::string argument() { std::cout << "--Making argument\n"; return "arg"; }

std::string v3errorPrepOld(std::string arg) {  //
    std::cout << "--Old error prep " << arg << std::endl;
    return arg;
}

std::string v3errorPrepNew(std::string arg) {  //
    lock();
    std::cout << "--New error prep " << arg << std::endl;
    return arg;
}

void v3errorEndOld(std::string arg) {  //
    std::cout << "--Old error end " << arg << std::endl;
}
void v3errorEndNew(std::string arg) {  //
    std::cout << "--New error end " << arg << std::endl;
    unlock();
}

int main() {
    {
        std::cout << "-----------\n";
        lock();
        v3errorEndOld(v3errorPrepOld(argument()));
    }
    {
        std::cout << "-----------\n";
        v3errorEndNew(v3errorPrepNew(argument()));
    }
    return 0;
}
```

Output:

```
--Lock
--Making argument
--Old error prep arg
--Old error end arg
-----------
--Making argument
--Lock
--New error prep arg
--New error end arg
--Unlock
```

The old code had a lock around building the arguments.